app/services/more_like_this_getter.rb

```diff
@@ -0,0 +1,87 @@
+# MoreLikeThisGetter.new(work).works
+#
+# returns an array of zero to ten
+# published works that SOLR thinks
+# are "like" the given work.
+#
+# This calls the SOLR server's more-like-this request handler
+# at http://SOME_URL/solr/SOME_CORE/mlt
+# to request a list of works that are deemed similar.
+#
+# See also:
+# work_indexer.rb for indexing details,
+# solr/config/solrconfig.xml for the request handler.
+#
+# We really do not want to slow down (or worse, break) the work show page
+# if SOLR is slow or unreachable, as it often is.
+# If anything goes wrong, you'll just get an empty array.
+#
+# MoreLikeThisGetter.new(work).works
+#
+class MoreLikeThisGetter
+  attr_reader :work
+
+  # These timeouts are short. We want to be conservative,
+  # as we don't want to trust our SOLR provider too much
+  # and this appears on a part of the website that's usually fast.
+  TIMEOUT=1
+  OPEN_TIMEOUT=1
+
+  # @param work [Work] Work
+  def initialize(work) 
+    @work = work
+  end
+
+  # Returns an array of up to 10 published works that SOLR deems similar.
+  def works
+    works_in_arbitrary_order = Work.where(
+      friendlier_id: friendlier_ids,
+      published: true)&.
+      index_by(&:friendlier_id) || {}
+    friendlier_ids.map {|id| works_in_arbitrary_order[id] }.compact
+  end
+
+  # Some justification for the choices SOLR made.
+  # Does not result in an additional call to SOLR.
+  def json
+    more_like_this_doc_set&.map do |doc|
+      doc.select do |key, value|
+        key == 'text1_tesim' || key.include?('more_like_this')
+      end
+    end
+  end
+
+  private
+
+  def friendlier_ids
+    @friendlier_ids ||= more_like_this_doc_set&.map { |d| d['id'] }
+  end
+
+  # Note: it appears that the mlt.count variable is ignored.
+  # mlt.qf causes a failure.
+  def mlt_params
+    @mlt_params ||= {
+      "q"         => "id:#{@work.friendlier_id}",
+      "mlt.fl"    => 'more_like_this_keywords_tsimv,more_like_this_fulltext_tsimv'
+    }
+  end
+
+  def solr_url
+    ScihistDigicoll::Env.lookup!(:solr_url)
+  end
+
+  def solr_connection
+    RSolr.connect( :url => solr_url, :timeout => TIMEOUT,:open_timeout => OPEN_TIMEOUT)
+  end
```

Comment:
OK, the hardest part... I wonder if there'a way to re-use the Solr connection that Blacklight is already using, intead of re-opening one. 

I am worried that re-opening a new one won't always get it quite right, like will it get all password and proxy and other settings we may apply to blacklight right? It'll also maybe be less performant to re-open a new connection every time instead of re-using one; not sure, this may not matter, depends on what Blacklight is doing. 

We need to spend more time reverse engineering Blacklight and trying to re-use as much as possible of it's code for where/how to make/get a Solr connection, and make sure we're doing it exactly the same way in case where we write our own code. This is a pain to do, yes, Blacklight is confusing. One part (although I don't understand where it's called how!) is at: https://github.com/projectblacklight/blacklight/blob/v7.29.0/lib/blacklight/solr/repository.rb#L108

Reply:
I think I have this figured out!